src/coreclr/jit/morph.cpp

```diff
@@ -13550,6 +13545,26 @@ GenTree* Compiler::fgOptimizeEqualityComparisonWithConst(GenTreeOp* cmp)
             DEBUG_DESTROY_NODE(rshiftOp->gtGetOp2());
             DEBUG_DESTROY_NODE(rshiftOp);
         }
+
+        // Optimizes (X & 1) == 1 to (X & 1)
+        // 
+        //                        EQ/NE                   AND
+        //                        /  \                   /  \.
+        //                      AND   CNS 1  ->         x   CNS 1
+        //                     /   \                  
+        //                    x   CNS 1
+        //           
+        // GTF_RELOP_JMP_USED is used to make sure the optimization is used for return statements only.
```

Comment:
```suggestion
        // The compiler requires jumps to have relop operands, so we do not fold that case.
```
Notably relops can appear anywhere r-values can, not just under `GT_RETURN`s.